src/luxonis_ml/data/dataset.py

```diff
@@ -502,6 +504,50 @@ def _add_process_batch():
                 )
                 self._end_time()
 
+            mask_paths = list(
+                set(
+                    [
+                        data["value"]
+                        for data in batch_data
+                        if data["type"] == "segmentation"
+                        or data["type"] == DataLabelType.SEGMENTATION
+                    ]
+                )
+            )
+            if len(mask_paths):
+                print("Checking masks...")
+                self._start_time()
+                mask_paths = data_utils.check_segmentation_masks(mask_paths)
+                self._end_time()
+                print("Generating mask UUIDs...")
+                self._start_time()
+                mask_uuid_dict = self.fs.get_file_uuids(
+                    mask_paths, local=True
+                )  # TODO: support from bucket
+                self._end_time()
+                if self.bucket_storage != BucketStorage.LOCAL:
+                    print("Uploading segmentation masks...")
+                    # TODO: support from bucket (likely with a self.fs.copy_dir)
+                    self._start_time()
+                    mask_upload_dict = self.fs.put_dir(
+                        local_paths=mask_paths,
+                        remote_dir="masks",
+                        uuid_dict=mask_uuid_dict,
+                    )
+                    self._end_time()
+                print("Finalizing paths...")
+                for data in tqdm(batch_data):
+                    if (
+                        data["type"] == "segmentation"
+                        or data["type"] == DataLabelType.SEGMENTATION
+                    ):
+                        if self.bucket_storage != BucketStorage.LOCAL:
+                            remote_path = mask_upload_dict[data["value"]]
+                            remote_path = f"{self.fs.protocol}://{os.path.join(self.fs.path, remote_path)}"
```

Comment:
Could we store local paths only, and prefix is defined somewhere to differentiate between LOCAL and remote? E.g., path would always be fixed to some degree, as the structure would be the same if on remote bucket or local, no?

Reply:
At the moment no, since in local mode (no bucket storage) we've stopped duplicating the data and copying files to some standard path. Instead, we load from the original provided filepaths. The idea was to make it more friendly for local development and not duplicate data. Thoughts?

Reply:
The way it's currently implemented is that we have this standardization for remote storage, so yes to the first bullet. 

But to the second we do not copy any data to a standard local path. Instead, it's up to the user to generate the masks and have them saved wherever and then provide paths to them. So you could theoretically have masks in different directories across your filesystem and it would still work.

Since we will likely switch to RLE, we could imagine the same for a voxel annotation. We would avoid copying the data locally, assuming the user already has it saved somewhere, and just point to it.

Reply:
Ok this makes sense, but wouldn't this make syncing a bit tricky? And if you push to bucket and pull from it, it would likely re-structure the things around no?

It still seems better than doing any copying though. Maybe a random idea and would have to be thought out more (not sure if it can work), but what about having this "standard path" per annotation type. So e.g. you could have custom path for segmentation still. It would still have to be standardized to some degree though, so not sure if worth it.